lib/mongoid/contextual/memory.rb

```diff
@@ -188,15 +188,16 @@ def limit(value)
         self
       end
 
+      # Pluck the field values in memory.
+      #
+      # @example Get the values in memory.
+      #   context.pluck(:name)
+      #
+      # @param [ String | Symbol ] *fields Field(s) to pluck.
+      #
+      # @return [ Array ] The array of plucked values.
       def pluck(*fields)
-        fields = Array.wrap(fields)
-        documents.map do |doc|
-          if fields.size == 1
-            doc[fields.first]
-          else
-            fields.map { |n| doc[n] }.compact
-          end
-        end.compact
+        documents.pluck(*fields)
```

Comment:
I wonder if this wasn't implemented like this originally because we supported versions of ruby that didn't implement pluck... do you know when pluck was made available?

Reply:
Yes, correct. Enumerable#pluck was [introduced in ActiveSupport 5.0.0](https://github.com/rails/rails/issues/20339)